### ISSUE TITLE
Referencing a VaultConnection CR from VaultAuth CR in another namespace does not work as documented

### ISSUE THREAD

**Issue by Stef16Robbe** [hashicorp/vault-secrets-operator#670]
**Describe the bug**
Trying to reference a `VaultConnection` CR from `VaultAuth` CR in another namespace does not work as documented. The [latest documentation](https://developer.hashicorp.com/vault/docs/platform/k8s/vso/api-reference#vaultauthspec) states the following:

```plaintext
VaultAuthRef to the VaultAuth resource, can be prefixed with a namespace, eg: namespaceA/vaultAuthRefB. If no namespace prefix is provided it will default to namespace of the VaultAuth CR.
```

But when I try this, I get a not found error in the `VaultAuth` status:

```yaml
status:
  error: VaultConnection.secrets.hashicorp.com "vaultconnection" not found
  valid: false
```

**Deployment files to Reproduce**

`vaultauth.yaml`:
```yaml
---
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultAuth
metadata:
  labels:
    app.kubernetes.io/name: vaultauth
    app.kubernetes.io/instance: vaultauth-sample
    app.kubernetes.io/part-of: vault-secrets-operator
    app.kubernetes.io/created-by: vault-secrets-operator
  name: vaultauth-sample
  namespace: app-test-dev
spec:
  vaultConnectionRef: vault-secrets-operator-system/vaultconnection # <-- this one!
  method: kubernetes
  mount: kubernetes
  kubernetes:
    role: test-role
    serviceAccount: vault-sa
```

`vaultconn.yaml`:
```yaml
---
apiVersion: secrets.hashicorp.com/v1beta1
kind: VaultConnection
metadata:
  labels:
    app.kubernetes.io/name: vaultconnection
    app.kubernetes.io/instance: vaultconnection-sample
    app.kubernetes.io/part-of: vault-secrets-operator
    app.kubernetes.io/created-by: vault-secrets-operator
  name: vaultconnection
  namespace: vault-secrets-operator-system # <-- inside operator namespace since I would like to use this for every application
spec:
  address: <url>
```

I am using Kustomize as you can see with the config provided in this repo: https://github.com/hashicorp/vault-secrets-operator/tree/main/config

**Expected behavior**
The `VaultConnection` CR created inside `vault-secrets-operator-system` namespace is found and can be referenced from another namespace, `app-test-dev` in this case.

**Environment**
* Kubernetes version: OpenShift

```
❯ oc version
...
Server Version: 4.13.36
Kubernetes Version: v1.26.13+8f85140
```

* vault-secrets-operator version: Controller image version: `registry.connect.redhat.com/hashicorp/vault-secrets-operator:0.5.2-ubi`

**Additional context**
It does work when the `VaultConnection` is present in the same namespace, and no namespace prefix is used.

I also tried giving the default sa a CR/CRB to view `VaultConnection` CRs in said namespace but to no avail.

**Comment by tvoran** [comment]
Hi @Stef16Robbe, so far I'm not able to duplicate this problem in OpenShift; referencing a VaultConnection in the operator's namespace from a VaultAuth in another namespace seems to be working in v0.5.2 and v0.6.0. I think there are cases where the VaultAuth's Status.Error isn't cleared when it passes validation if it failed previously, but since yours shows `Valid: false` that's probably not it.

Does `kubectl describe -n vault-secrets-operator-system vaultconnection` show any issues in the status or events? Do you see any errors or warning in the operator's logs related to this?

**Comment by Stef16Robbe** [comment]
Hi @tvoran apologies for the late reply, a bunch of stuff got in the way.

Anyways I have good news, I've managed to make it work, using almost the same config I posted above. We wiped our sandbox cluster and instead of copying the installation manifests from [here](https://github.com/hashicorp/vault-secrets-operator/tree/main/config) I decided to install the operator using the openshift operatorhub, and now it all works fine.

That's the only relevant change I've made, so I'm not sure what the example manifests might lack, but from my end this can be closed.
